lib/libc/math/lib_logf.c

```diff
@@ -81,17 +83,25 @@ float logf(float x)
 	iter = 0;
 	relax_factor = 1;
 
+	if (x < 0.0) {
+                return NAN;
+        }
+
+	if (x == 0.0) {
+                return -HUGE_VAL;
+        }
+
 	while (y > y_old + epsilon || y < y_old - epsilon) {
 		y_old = y;
 		ey = expf(y);
 		y -= (ey - x) / ey;
 
-		if (y > 700.0) {
-			y = 700.0;
+		if (y > FLT_MAX_EXP_X) {
```

Comment:
The original r-value is 700.0, but you changed to 88.0F.
Can you tell me why did you change like this?

Reply:
Max value of y in logf() is 88.722839 which will be round off to 88.0F same as NuttX logf().